pkg/privilege/privileges/privileges.go

```diff
@@ -77,13 +77,17 @@ type UserPrivileges struct {
 	host string
 	*Handle
 	extensionAccessCheckFuncs []extension.AccessCheckFunc
+	authPlugins               map[string]*extension.AuthPlugin
+	userRecord                *UserRecord
+	sessionVars               *variable.SessionVars
```

Comment:
These fields are too complex and it's hard to figure out their usage from the field names. We only need to store some plugin-related auth context here, how about just add two functions like `pluginRequestVerification` and `pluginRequestDynamicVerfication` here ?